enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java

```diff
@@ -57,25 +61,33 @@
     private final Map<String,WritableFulltext> writableNodeIndices;
     private final Map<String,WritableFulltext> writableRelationshipIndices;
     private final FulltextUpdateApplier applier;
+    private final FulltextFactory factory;
     private final ReadWriteLock configurationLock;
 
     /**
      * Creates a provider of fulltext indices for the given database. This is the entry point for all fulltext index operations.
+     *
+     *
      * @param db Database that this provider should work with.
      * @param log For logging errors.
      * @param availabilityGuard Used for waiting with populating the index until the database is available.
      * @param scheduler For background work.
      * @param transactionIdStore Used for checking if the store has had transactions applied to it, while the fulltext
-     * indexes have been disabled. If so, then the indexes will be rebuilt.
+     * @param fileSystem The filesystem to use.
+     * @param storeDir Store directory of the database.
+     * @param analyzerClassName The Lucene analyzer to use for the {@link LuceneFulltext} created by this factory.
      */
     public FulltextProvider( GraphDatabaseService db, Log log, AvailabilityGuard availabilityGuard,
-                             JobScheduler scheduler, TransactionIdStore transactionIdStore )
+                             JobScheduler scheduler, TransactionIdStore transactionIdStore,
+                             FileSystemAbstraction fileSystem, File storeDir,
+                             String analyzerClassName ) throws IOException
```

Comment:
I'm not convinced that the provider should be tied to one specific analyzer?

Reply:
In the future I want that parameter to be understood as the _default_ analyser, and provide an overload of `createIndex` that also takes an analyser.

Reply:
I see, that makes sense